app/models/capture.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const db      = require("../../app/db");
+const Promise = require("bluebird");
+
+const CommConns = require("../models/commConns");
+
+class CaptureBoard {
+
+  static findByOrg (orgId) {
+    return new Promise((fulfill, reject) => {
+      db("msgs")
+        .leftJoin("convos", "msgs.convo", "convos.convid")
+        .leftJoin("comms", "comms.commid", "msgs.comm")
+        .leftJoin("commconns", "commconns.commconnid", "comms.commid")
+        .where("convos.client", null)
+        .andWhere("convos.open", true)
+        .orderBy("msgs.created", "ASC")
+      .then((floaters) => {
+
+        // Reduce results to just convids, and sort incrementally
+        let convos = floaters.map((ea) => { 
+          return ea.convo; 
+        }).reduce((a,b) => { 
+          if (a.indexOf(b) < 0) {
+            a.push(b);
+          }
+          return a;
+        }, []).map((ea) => { 
+          return {
+            convo: ea, 
+            msgs: []
+          };
+        });
+
+        // Add messages to each identified convo obj
+        floaters.forEach((ea) => { 
+          for (let i = 0; i < convos.length; i++) {
+            if (convos[i].convo == ea.convo) convos[i].msgs.push(ea);
+          }
+        });
+
+        fulfill(convos)
+      }).catch(reject);
+    })
+  }
+
+  static findByConversationId (orgId, conversationId) {
+    return new Promise((fulfill, reject) => {
+      CaptureBoard.findByOrg(orgId)
+      .then((conversations) => {
+        let conversation = conversations.filter((convo) => {
+          return conversationId == convo.convo;
+        })[0];
+        fulfill(conversation)
+      }).catch(reject);
+    })
+  }
+
+  static associateConversation (user, client, conversationId) {
+    return new Promise((fulfill, reject) => {
+      let subject = "Captured Conversation from New Contact";
+      let deviceName = "New Captured Method";
+      
+      // Query 1: Update convo with CM and client
+      db("convos")
```

Comment:
yeah this is a sort of monstrous function that is translated directly from the older version of the application... will clean up later but for now i just stuck in the fulfill and reject actions and called it a day to get the functionality transferred over